packages/widgets/src/Transfer/Transfer.js

```diff
@@ -108,6 +119,16 @@ export const Transfer = ({
     const actualFilter = onFilterChange ? searchTerm : internalFilter
     const actualFilterCallback = filterable ? filterCallback : identity
 
+    const [internalFilterPicked, setInternalFilterPicked] = useState(
+        initialSearchTermPicked
+    )
+    const actualFilterPicked = onFilterPickedChange
+        ? searchTermPicked
+        : internalFilterPicked
+    const actualFilterPickedCallback = filterablePicked
+        ? filterCallbackPicked
+        : identity
```

Comment:
It's a bit unclear what's going on in this area.

Reply:
I've tried to make the code a bit more clear. Is it better now?